pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/AuthenticationTest.java

```diff
@@ -1,110 +1,182 @@
+    }
 
-        Patch patch = new Patch("0", "participantId", "0", gpUserId , new NameValue("oD.locationPx",  "location"), null, "XSZSRS1MS3D4OAEK2DPM") ;
+    @Test
+    public void getKitShipperRoleCannotChangeParticipantInfo(){
+        String realm = ddpInstance.getInstanceName();
+
+        Patch patch = new Patch("0", "participantId", "0", cmiKitShippingOnlyUser.getIdAsString() , new NameValue("oD.locationPx",  "location"), null, "XSZSRS1MS3D4OAEK2DPM") ;
         patch.setTableAlias("oD");
 
-        Assert.assertFalse(UserUtil.checkKitShippingAccessForPatch(realm, gpUserId, null, patch));
+        Assert.assertFalse(UserUtil.checkKitShippingAccessForPatch(realm, cmiKitShippingOnlyUser.getIdAsString(), null, patch));
     }
 
+    /**
+     * Verifies that users who don't have permission to make
+     * patch requests can't make patch requests, and that
+     * users who do have said permission can make the requests.
+     */
     @Test
-    @Ignore
     public void mismatchAccessTest(){
-        String realm = "osteo2";
-
-        String userId = Integer.toString(new UserDao().getUserByEmail(cfg.getString(UNIT_TESTER_EMAIL)).orElse(new UserDto()).getId());
-        String gpUserId = Integer.toString(new UserDao().getUserByEmail(cfg.getString(GP_UNIT_TESTER_EMAIL)).orElse(new UserDto()).getId() );
+        String realm = ddpInstance.getInstanceName();
 
-        Patch patch1 = new Patch("0", "participantId", "0", userId , new NameValue("oD.locationPx",  "location"), null, "XSZSRS1MS3D4OAEK2DPM") ;
+        Patch patch1 = new Patch("0", "participantId", "0", dsmAdminUser.getIdAsString() , new NameValue("oD.locationPx",  "location"), null, "XSZSRS1MS3D4OAEK2DPM") ;
         patch1.setTableAlias("oD");
 
         try {
-            Assert.assertFalse(UserUtil.checkUserAccessForPatch(realm, gpUserId, DBConstants.PT_LIST_VIEW, null, patch1));
+            UserUtil.checkUserAccessForPatch(realm, cmiKitShippingOnlyUser.getIdAsString(), PT_LIST_VIEW, null, patch1);
+            Assert.fail("Did not throw expected exception");
         } catch (Exception e) {
             Assert.assertTrue(e.getMessage().contains("User id in patch did not match the one in token"));
         }
         try {
-            Assert.assertFalse(UserUtil.checkUserAccessForPatch(realm, gpUserId, DBConstants.MR_VIEW, null, patch1));
+            UserUtil.checkUserAccessForPatch(realm, cmiKitShippingOnlyUser.getIdAsString(), DBConstants.MR_VIEW, null, patch1);
+            Assert.fail("Did not throw expected exception");
         } catch (Exception e) {
             Assert.assertTrue(e.getMessage().contains("User id in patch did not match the one in token"));
         }
         try {
-            Assert.assertFalse(UserUtil.checkUserAccessForPatch(realm, gpUserId, DBConstants.MR_ABSTRACTER, null, patch1));
+            UserUtil.checkUserAccessForPatch(realm, cmiKitShippingOnlyUser.getIdAsString(), DBConstants.MR_ABSTRACTER, null, patch1);
```

Comment:
Since `checkUserAccessForPatch` is expected to throw an exception here, I remove the asserts--which are never run--and added explicit `fail()` in case the exceptions aren't thrown.